src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,112 @@
+                book.setId(id);
+            }
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book wasn't added. 0 rows were affected");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error while adding book", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SELECT_BY_ID)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getBookFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred wile finding by id", e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SELECT_ALL)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> res = new ArrayList<>();
+            while (resultSet.next()) {
+                res.add(getBookFromResultSet(resultSet));
+            }
+            return res;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred while getting all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(UPDATE)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book wasn't changed. 0 rows were affected");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred while updating book", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred while deleting book", e);
+        }
+    }
+
+    private static Book getBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Long bookId = resultSet.getLong("id");
```

Comment:
it is better to use `getObject` here, since `getLong` will return 0 if id is missing